src/Navigation/Plugins/recastJSPlugin.ts

```diff
@@ -0,0 +1,284 @@
+import { INavigationEnginePlugin, ICrowd, AgentParameters, NavMeshParameters } from "../../Navigation/INavigationEngine";
+import { Logger } from "../../Misc/logger";
+import { VertexData } from "../../Meshes/mesh.vertexData";
+import { AbstractMesh } from "../../Meshes/abstractMesh";
+import { Mesh } from "../../Meshes/mesh";
+import { Scene } from "../../scene";
+import { Vector3 } from '../../Maths/math';
+import { TransformNode } from "../../Meshes/transformNode";
+
+declare var Recast: any;
+
+/**
+ * RecastJS navigation plugin
+ */
+export class RecastJSPlugin implements INavigationEnginePlugin {
+    /**
+     * Reference to the Recast library
+     */
+    public bjsRECAST: any = {};
+    public name: string = "RecastJSPlugin";
+    public navMesh: any;
+
+    /**
+     * Initializes the recastJS plugin
+     */
+    public constructor(recastInjection: any = Recast) {

```

Comment:
pinging @sebavan to make sure this is ok with modules

Reply:
Seen with Seb and fixed.